test/unit/Roave/InfectionStaticAnalysisTest/CliUtilityTest.php

```diff
@@ -0,0 +1,150 @@
+
+    /**
+     * @param list<non-empty-string> $expectedNewArguments
+     * @param non-empty-string|null  $expectedArgumentValue
+     * @param list<non-empty-string> $arguments
+     * @param non-empty-string       $argument
+     *
+     * @dataProvider provideExtractionData
+     */
+    public function testExtractArgument(
+        array $expectedNewArguments,
+        ?string $expectedArgumentValue,
+        array $arguments,
+        string $argument
+    ): void {
+        $result = CliUtility::extractArgument($arguments, $argument);
+
+        self::assertSame($expectedNewArguments, $result[0]);
+        self::assertSame($expectedArgumentValue, $result[1]);
+    }
+
+    /**
+     * @return list<array{0: (non-empty-string|null), 1: list<non-empty-string>, 2: non-empty-string}>
+     */
+    public function provideExtractionData(): array
+    {
+        return [
+            [
+                ['vendor/bin/roave-infection-static-analysis-plugin'],
+                'configuration/psalm.xml',
+                ['vendor/bin/roave-infection-static-analysis-plugin', '--psalm-config', 'configuration/psalm.xml'],
+                'psalm-config',
+            ],
+            [
+                ['vendor/bin/roave-infection-static-analysis-plugin'],
+                'configuration/psalm.xml',
+                ['vendor/bin/roave-infection-static-analysis-plugin', '--psalm-config', '"configuration/psalm.xml"'],
+                'psalm-config',
+            ],
+            [
+                ['vendor/bin/roave-infection-static-analysis-plugin'],
+                'configuration/psalm.xml',
+                ['vendor/bin/roave-infection-static-analysis-plugin', '--psalm-config="configuration/psalm.xml"'],
+                'psalm-config',
+            ],
+            [
+                ['vendor/bin/roave-infection-static-analysis-plugin'],
+                'configuration/psalm.xml',
+                ['vendor/bin/roave-infection-static-analysis-plugin', '--psalm-config=configuration/psalm.xml"'],
+                'psalm-config',
+            ],
+            [
+                ['vendor/bin/roave-infection-static-analysis-plugin'],
+                'configuration/psalm.xml',
+                ['vendor/bin/roave-infection-static-analysis-plugin', '--psalm-config="configuration/psalm.xml'],
+                'psalm-config',
+            ],
+            [
+                ['vendor/bin/roave-infection-static-analysis-plugin'],
+                'configuration/psalm.xml',
+                ['vendor/bin/roave-infection-static-analysis-plugin', '--psalm-config=configuration/psalm.xml'],
+                'psalm-config',
+            ],
+            [
+                ['vendor/bin/roave-infection-static-analysis-plugin', '--psalm-config=foo'],
+                'configuration/psalm.xml',
+                ['vendor/bin/roave-infection-static-analysis-plugin', '--psalm-config=configuration/psalm.xml', '--psalm-config=foo'],
+                'psalm-config',
+            ],
```

Comment:
should we extract both `--psalm-config` and return an array that we can later iterate over? ( possible directories? ).

I don't think it would be any useful to actually take more than 1 path into consideration, but assuming someone provided `--psalm-config` twice, Symfony console will later throw because of the unknown  `--psalm-config` argument.

so the options:

1. keep as is, to not add complexity to the CliUtility class, as this is really an edge case.
2. extract all possible values, and return all of them ( then we can iterate over the array if multiple values are provided )
3. remove additional values from the arguments list, but only return the first one.

Reply:
the reason I asked for the code to be extracted to a utility class is just for testing purposes: by all means, it could've been hardcoded to only deal with `--psalm-config`, and nothing else. It is `@internal` anyway 👍